src/components/FullscreenHandler/index.js

```diff
@@ -0,0 +1,50 @@
+import { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+import { PROP_TYPE_CHILDREN } from '../constants'
+import { renderChildren } from '../helpers'
+
+const fullscreenEvents = [
+  'fullscreenchange',
+  'mozfullscreenchange',
+  'webkitfullscreenchange',
+  'msfullscreenchange',
+]
+
+export default class FullscreenHandler extends PureComponent {
+  static propTypes = {
+    onFullscreenChange: PropTypes.func,
+    children: PROP_TYPE_CHILDREN.isRequired,
+  }
+
+  state = {
+    fullscreenElement: undefined,
+  }
+
+  componentDidMount () {
+    this.checkFullscreenState()
+    fullscreenEvents.forEach(eventName => document.addEventListener(
+      eventName, this.checkFullscreenState))
+  }
+
+  componentWillUnmount () {
+    fullscreenEvents.forEach(eventName => document.removeEventListener(
+      eventName, this.checkFullscreenState))
+  }
+
+  checkFullscreenState = () => {
+    const fullscreenElement = document.fullscreenElement
+      || document.mozFullScreenElement
+      || document.webkitFullscreenElement
+      || document.msFullscreenElement
+
+    this.setState({ fullscreenElement })
+  }
+
+  render () {
+    const { children } = this.props
+    const { fullscreenElement } = this.state
+
+    return renderChildren(children, { fullscreenElement })
```

Comment:
i wonder if it might be beneficial to pass along another prop that is the fullscreenElement when in fullscreen, and document.body when not

Reply:
heard that - what would you call that, though?  currentRootNode?

Reply:
the more I think about it though, feels a bit out-of-bounds

Reply:
let's sit on that idea for a bit before moving on it.  but i was thinking something like `screenElement`